src/actions/editorialWorkflow.js

```diff
@@ -291,3 +326,66 @@ export function publishUnpublishedEntry(collection, slug) {
     });
   };
 }
+
+export const registerUnpublishedEntryDependency = unpublishedEntryRegisterDependency;
+
+export function getUnpublishedEntryDependencies (collection, slug) {
+  return (dispatch, getState) => {
+    dispatch(unpublishedEntryDependenciesRequest(collection, slug));
+
+    const getDepsPath = dep => [
+      "entities",
+      dep,
+      "metaData",
+      "dependencies",
+    ];
+
+    const getEventualDependencies = (paths, loadedDeps, state) =>
```

Comment:
This function is only useful within `getUnpublishedEntryDependencies` - I didn't want it exposed to the outer scope since it's very specific to the functionality of `traverse`. I'm fine with extracting it, that was just the reasoning behind nesting it with `traverse` (which can't be hoisted).